

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Make .mti File &mdash; Motsi Tutorial v0.1.21 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=18f8a353"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            Motsi Tutorial
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Make <code class="docutils literal notranslate"><span class="pre">.mti</span></code> File</a><ul>
<li><a class="reference internal" href="#mtirna">mtiRNA</a></li>
<li><a class="reference internal" href="#mtiepi">mtiEpi</a></li>
<li><a class="reference internal" href="#mtitf">mtiTf</a></li>
<li><a class="reference internal" href="#mtiepihic">mtiEpiHiC</a></li>
</ul>
</li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Motsi Tutorial</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Make <code class="docutils literal notranslate"><span class="pre">.mti</span></code> File</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/1.make_mti_file.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="make-mti-file">
<h1>Make <code class="docutils literal notranslate"><span class="pre">.mti</span></code> File<a class="headerlink" href="#make-mti-file" title="Link to this heading"></a></h1>
<p>The <code class="docutils literal notranslate"><span class="pre">.mti</span></code> format is the file format used by <strong>Motsi</strong> to store time-series data. The first line is the Motsi header line, and the second line is the Title.</p>
<p>The columns include base columns that record positional information and value columns that record time-series data information.</p>
<p>Creating an <code class="docutils literal notranslate"><span class="pre">.mti</span></code> format file is the first step in using <strong>Motsi</strong>, as it serves as the input file for other <strong>Motsi</strong> analysis functions.</p>
<p><strong>Motsi</strong> can conveniently generate <code class="docutils literal notranslate"><span class="pre">.mti</span></code> format files for various types of omics data, including transcriptome data (e.g., RNA-seq), epigenome data (e.g., Histone ChIP-seq), transcription factor data (e.g., TF ChIP-seq), and 3D epigenome data (e.g., Hi-C - ChIP-seq).</p>
<p>Next, I will guide you through a Quick Start to help you quickly master the usage of these functions:</p>
<section id="mtirna">
<h2>mtiRNA<a class="headerlink" href="#mtirna" title="Link to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">mtiRNA</span></code> is a method provided by <strong>Motsi</strong> for generating <code class="docutils literal notranslate"><span class="pre">.mti</span></code> files from RNA-seq data. It is developed based on <strong>Kallisto</strong>, uses fastq files as input, and can quickly calculate the TPM values of target genes without requiring time-consuming alignment.</p>
<p>Before using <code class="docutils literal notranslate"><span class="pre">mtiRNA</span></code>, <strong>Kallisto</strong> needs to be installed. We recommend using <strong>conda</strong> for a quick setup:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>install<span class="w"> </span>kallisto
</pre></div>
</div>
<p>Then you can use <code class="docutils literal notranslate"><span class="pre">mtiRNA</span></code> through：</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="w"> </span>motsi.py<span class="w"> </span>mtiRNA<span class="w"> </span><span class="o">[</span>-h<span class="o">]</span><span class="w"> </span>-f<span class="w"> </span>FASTQ<span class="w"> </span><span class="o">[</span>FASTQ<span class="w"> </span>...<span class="o">]</span><span class="w"> </span>-g<span class="w"> </span>GENE<span class="w"> </span>-i<span class="w"> </span>INDEX<span class="w"> </span>-gtf<span class="w"> </span>GTF<span class="w"> </span><span class="o">[</span>-f2<span class="w"> </span>FASTQ<span class="w"> </span><span class="o">[</span>FASTQ<span class="w"> </span>...<span class="o">]]</span>
<span class="w">                       </span><span class="o">[</span>-s<span class="w"> </span>SAMPLE<span class="w"> </span><span class="o">[</span>SAMPLE<span class="w"> </span>...<span class="o">]]</span><span class="w"> </span><span class="o">[</span>-t<span class="w"> </span>THREAD<span class="o">]</span><span class="w"> </span><span class="o">[</span>--norm_type<span class="w"> </span><span class="o">{</span>agmn,tpm<span class="o">}]</span><span class="w"> </span><span class="o">[</span>-p<span class="w"> </span>PREFIX<span class="o">]</span>
</pre></div>
</div>
<p>required arguments:
<code class="docutils literal notranslate"><span class="pre">-f</span></code>: List of input .fastq files. For paired-end data, provide _1 file here
<code class="docutils literal notranslate"><span class="pre">-g</span></code>: Gene bed6 file.
<code class="docutils literal notranslate"><span class="pre">-i</span></code>: Kallisto index file.</p>
<p>​	– You can make it by <code class="docutils literal notranslate"><span class="pre">$</span> <span class="pre">kallisto</span> <span class="pre">index</span></code>.
<code class="docutils literal notranslate"><span class="pre">-gtf</span></code> : <code class="docutils literal notranslate"><span class="pre">.gtf</span></code> file. Here is some example to show you how to download it:</p>
<p>​	– hg38: <code class="docutils literal notranslate"><span class="pre">wget</span> <span class="pre">-c</span> <span class="pre">https://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz</span></code></p>
<p>​	– hg19: <code class="docutils literal notranslate"><span class="pre">wget</span> <span class="pre">-c</span> <span class="pre">ftp://ftp.ensembl.org/pub/release-75/gtf/homo_sapiens/Homo_sapiens.GRCh37.75.gtf.gz</span></code></p>
<p>options:
<code class="docutils literal notranslate"><span class="pre">-f2</span></code>:  List of input _2 .fastq files. Only usable for pair-end sequencing data. (Default: None)
<code class="docutils literal notranslate"><span class="pre">-s</span></code>: List of sample names. (Default: T1,T2,…)
<code class="docutils literal notranslate"><span class="pre">-t</span></code> : Number of threads. (Default: 1)
<code class="docutils literal notranslate"><span class="pre">--norm_type</span></code> : Normalization type.  (Default: tpm).  <code class="docutils literal notranslate"><span class="pre">agmn</span></code> for all gene median normalization
<code class="docutils literal notranslate"><span class="pre">-p</span></code>: Prefix for output file. (Default: MotsiRNAResults)</p>
</section>
<section id="mtiepi">
<h2>mtiEpi<a class="headerlink" href="#mtiepi" title="Link to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">mtiEpi</span></code> can generate <code class="docutils literal notranslate"><span class="pre">.mti</span></code> files from aligned epigenetics data as input. We recommend using <code class="docutils literal notranslate"><span class="pre">.bigWig</span></code>format, as it is the fastest input format for processing. Additionally, <code class="docutils literal notranslate"><span class="pre">.bam</span> </code>and <code class="docutils literal notranslate"><span class="pre">.bedGraph</span></code> files are also supported as input.</p>
<p><code class="docutils literal notranslate"><span class="pre">mtiEpi</span></code> is developed based on <strong>Tichr</strong>, so before using it, please ensure that <strong>bedtools</strong> (a prerequisite for <strong>Tichr</strong>) is available in your environment. If you haven’t installed <strong>bedtools</strong> yet, you can quickly install it via <strong>conda</strong> using the following command:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>install<span class="w"> </span>-c<span class="w"> </span>bioconda<span class="w"> </span>bedtools
</pre></div>
</div>
<p>Then you can use <code class="docutils literal notranslate"><span class="pre">mtiEpi</span></code> through：</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>motsi.py<span class="w"> </span>mtiEpi<span class="w"> </span><span class="o">[</span>-h<span class="o">]</span><span class="w"> </span>-f<span class="w"> </span>FILE<span class="w"> </span><span class="o">[</span>FILE<span class="w"> </span>...<span class="o">]</span><span class="w"> </span>-gt<span class="w"> </span>GENOME_TABLE<span class="w"> </span>-g<span class="w"> </span>GENE<span class="w"> </span><span class="o">[</span>-s<span class="w"> </span>SAMPLE<span class="w"> </span><span class="o">[</span>SAMPLE<span class="w"> </span>...<span class="o">]]</span><span class="w"> </span><span class="o">[</span>-p<span class="w"> </span>PREFIX<span class="o">]</span>
<span class="w">                       </span><span class="o">[</span>--norm_type<span class="w"> </span><span class="o">{</span>no_norm,agmn,spmr<span class="o">}]</span><span class="w"> </span><span class="o">[</span>--file_type<span class="w"> </span><span class="o">{</span>bw,bedg,bam<span class="o">}]</span><span class="w"> </span><span class="o">[</span>-max_distance<span class="w"> </span>MAXDISTANCE<span class="o">]</span>
</pre></div>
</div>
<p>required arguments:
<code class="docutils literal notranslate"><span class="pre">-f</span></code> : List of input files.
<code class="docutils literal notranslate"><span class="pre">-gt</span></code> : Genome table file. Records chromosome length.
<code class="docutils literal notranslate"><span class="pre">-g</span></code>: Gene bed6 file.</p>
<p>options:
<code class="docutils literal notranslate"><span class="pre">-s</span></code>: List of sample names. (Default: T1,T2,…)
<code class="docutils literal notranslate"><span class="pre">-p</span></code>: Prefix for output file. (Default: MotsiEpiResults)
<code class="docutils literal notranslate"><span class="pre">--norm_type</span></code>:  Normalization type. {no_norm,agmn,spmr} (Default: agmn)
<code class="docutils literal notranslate"><span class="pre">--file_type</span></code>: File type for input files. {bw,bedg,bam}  (Default: bw)
<code class="docutils literal notranslate"><span class="pre">-max_distance</span></code>: Max distance from site to gene TSS. For a given gene, epigenetic signals beyond this distance will not be considered.(Default: 10000)</p>
</section>
<section id="mtitf">
<h2>mtiTf<a class="headerlink" href="#mtitf" title="Link to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">mtiTf</span></code> is a function designed for transcription factor analysis. It takes transcription factor ChIP-seq/Cut&amp;Tag data as input to generate <code class="docutils literal notranslate"><span class="pre">.mti</span></code> files. Due to the diverse binding sites of transcription factors, <code class="docutils literal notranslate"><span class="pre">mtiTf</span></code> requires an additional candidate_site file in bed3 format that records potential TF binding sites, unlike <code class="docutils literal notranslate"><span class="pre">mtiEpi</span></code>.</p>
<p>The usage of mtiTf is as follows:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="w"> </span>motsi.py<span class="w"> </span>mtiTf<span class="w"> </span><span class="o">[</span>-h<span class="o">]</span><span class="w"> </span>-f<span class="w"> </span>FILE<span class="w"> </span><span class="o">[</span>FILE<span class="w"> </span>...<span class="o">]</span><span class="w"> </span>-gt<span class="w"> </span>GENOME_TABLE<span class="w"> </span>-c<span class="w"> </span>CANDIDATE_SITE<span class="w"> </span><span class="o">[</span>-s<span class="w"> </span>SAMPLE<span class="w"> </span><span class="o">[</span>SAMPLE<span class="w"> </span>...<span class="o">]]</span>
<span class="w">                      </span><span class="o">[</span>-p<span class="w"> </span>PREFIX<span class="o">]</span><span class="w"> </span><span class="o">[</span>--norm_type<span class="w"> </span><span class="o">{</span>no_norm,agmn,spmr<span class="o">}]</span><span class="w"> </span><span class="o">[</span>--file_type<span class="w"> </span><span class="o">{</span>bw,bedg,bam<span class="o">}]</span>
</pre></div>
</div>
<p>required arguments:
<code class="docutils literal notranslate"><span class="pre">-f</span></code> : List of input files.
<code class="docutils literal notranslate"><span class="pre">-gt</span></code> : Genome table file. Records chromosome length.
<code class="docutils literal notranslate"><span class="pre">-g</span></code>: Gene bed6 file.
<code class="docutils literal notranslate"><span class="pre">-c</span></code>： Candidate site file – a file in BED3 format that records potential transcription factor binding sites.</p>
<p>options:
<code class="docutils literal notranslate"><span class="pre">-s</span></code>: List of sample names. (Default: T1,T2,…)
<code class="docutils literal notranslate"><span class="pre">-p</span></code>: Prefix for output file. (Default: MotsiEpiResults)
<code class="docutils literal notranslate"><span class="pre">--norm_type</span></code>:  Normalization type. {no_norm,agmn,spmr} (Default: agmn)
<code class="docutils literal notranslate"><span class="pre">--file_type</span></code>: File type for input files. {bw,bedg,bam}  (Default: bw)</p>
</section>
<section id="mtiepihic">
<h2>mtiEpiHiC<a class="headerlink" href="#mtiepihic" title="Link to this heading"></a></h2>
<p>Under development</p>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Puxuan Sun.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>