

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Differential Analysis &mdash; Motsi Tutorial v0.1.21 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=18f8a353"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Make .mti File" href="1.make_mti_file.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            Motsi Tutorial
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="1.make_mti_file.html">Make .mti File</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Differential Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#all-target-tsv"><strong>all_{target}.tsv</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#up-target-tsv-down-target-tsv">up{target}.tsv / down{target}.tsv</a></li>
<li class="toctree-l3"><a class="reference internal" href="#all-up-down-target-heatmap-pdf">All/Up/Down{target}_Heatmap.pdf</a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Motsi Tutorial</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Differential Analysis</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/2.differential_analysis.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="differential-analysis">
<h1>Differential Analysis<a class="headerlink" href="#differential-analysis" title="Link to this heading"></a></h1>
<p><code class="docutils literal notranslate"><span class="pre">Diff</span></code>is a differential analysis function provided by <strong>Motsi</strong>. By inputting different <code class="docutils literal notranslate"><span class="pre">.mti</span></code> files, you can use it to:</p>
<ul class="simple">
<li><p>Identify differentially expressed genes (Dynamic DEGs) in time-series data</p></li>
<li><p>Detect dynamically changing transcription factor binding sites</p></li>
<li><p>Discover dynamically changing cis-regulatory elements</p></li>
<li><p>Locate genomic sites with dynamic changes in 3D chromatin structure</p></li>
<li><p>And any other applications you can imagine!</p></li>
</ul>
<p><code class="docutils literal notranslate"><span class="pre">Diff</span></code> calculates a regulatory score for each genomic locus by multiplying a time-accumulated regulatory score by a noise coefficient. It then determines the statistical confidence of the differential analysis using a Gaussian distribution model based on multiple experimental replicates, ultimately computing an adjusted regulatory score.</p>
<p>If you have already obtained <code class="docutils literal notranslate"><span class="pre">.mti</span></code>files through the previous steps, you can perform <code class="docutils literal notranslate"><span class="pre">Diff</span></code> analysis quickly and easily using the following method:</p>
<div class="highlight-Shell notranslate"><div class="highlight"><pre><span></span>motsi.py<span class="w"> </span>Diff<span class="w"> </span><span class="o">[</span>-h<span class="o">]</span><span class="w"> </span>-m<span class="w"> </span>MTI<span class="w"> </span><span class="o">[</span>MTI<span class="w"> </span>...<span class="o">]</span><span class="w"> </span><span class="o">[</span>-t<span class="w"> </span>THRESHOLD<span class="o">]</span><span class="w"> </span><span class="o">[</span>-o<span class="w"> </span>OUTPUT<span class="o">]</span><span class="w"> </span><span class="o">[</span>--fillna<span class="w"> </span>FILL_NA<span class="o">]</span><span class="w"> </span><span class="o">[</span>--rep_weight<span class="w"> </span>REP_WEIGHT<span class="o">]</span>
<span class="w">                     </span><span class="o">[</span>--input_value<span class="w"> </span>INPUT_VALUE<span class="o">]</span>
</pre></div>
</div>
<p>required arguments:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-m</span></code> : List of input .mti files. Multiple files represent multiple replications.</p></li>
</ul>
<p>options:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-t</span></code>. Stable Differential Gene STFC thresholds (Default: 0.5)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>: Output dir. (Default: DifferentialResult)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--fillna</span></code> : Fill NA value with 0. (Default: True)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--rep_weight</span></code>: The weight for replications when calculating STD (Default: 0.5)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--input_value</span></code>: You can choose between inputting RawValues and Log2FoldChange. {raw,LFC}  (Default: raw)</p></li>
</ul>
<section id="output">
<h2>Output<a class="headerlink" href="#output" title="Link to this heading"></a></h2>
<p>After running <code class="docutils literal notranslate"><span class="pre">Diff</span></code>, you will obtain the following files:</p>
<section id="all-target-tsv">
<h3><strong>all_{target}.tsv</strong><a class="headerlink" href="#all-target-tsv" title="Link to this heading"></a></h3>
<p>A data table containing the computational results for all targets. The columns mean:</p>
<ul class="simple">
<li><p>base columns: Target information</p></li>
<li><p>value columns: The mean value between all replications.</p></li>
<li><p>STS: The raw regulation score for each replication.</p></li>
<li><p>UpPvalue: The p-value for considering a target as an <strong>Up-regulated Target</strong></p></li>
<li><p>DownPvalue: The p-value for considering a target as an <strong>Down-regulated Target</strong></p></li>
<li><p><em>AllMark</em>: A score used for ranking all targets, including both Up-regulated and Down-regulated ones. It is used in the heatmap generated by this function.</p></li>
</ul>
</section>
<section id="up-target-tsv-down-target-tsv">
<h3>up{target}.tsv / down{target}.tsv<a class="headerlink" href="#up-target-tsv-down-target-tsv" title="Link to this heading"></a></h3>
<p>The columns are identical to those in <code class="docutils literal notranslate"><span class="pre">all_{target}.tsv</span></code>. They respectively contain the filtered Up-regulated and Down-regulated targets.</p>
</section>
<section id="all-up-down-target-heatmap-pdf">
<h3>All/Up/Down{target}_Heatmap.pdf<a class="headerlink" href="#all-up-down-target-heatmap-pdf" title="Link to this heading"></a></h3>
<p>Heatmaps used to visualize the three aforementioned data tables. The values displayed represent the average of all replicate input values. These averages are transformed by calculating <code class="docutils literal notranslate"><span class="pre">Log10(value</span> <span class="pre">+</span> <span class="pre">1)</span></code>, followed by Z-score normalization for each target (i.e., each row).</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="1.make_mti_file.html" class="btn btn-neutral float-left" title="Make .mti File" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Puxuan Sun.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>