# Differential Analysis

`Diff`is a differential analysis function provided by **Motsi**. By inputting different `.mti` files, you can use it to:

- Identify differentially expressed genes (Dynamic DEGs) in time-series data
- Detect dynamically changing transcription factor binding sites
- Discover dynamically changing cis-regulatory elements
- Locate genomic sites with dynamic changes in 3D chromatin structure
- And any other applications you can imagine!

`Diff` calculates a regulatory score for each genomic locus by multiplying a time-accumulated regulatory score by a noise coefficient. It then determines the statistical confidence of the differential analysis using a Gaussian distribution model based on multiple experimental replicates, ultimately computing an adjusted regulatory score.

If you have already obtained `.mti`files through the previous steps, you can perform `Diff` analysis quickly and easily using the following method:

```Shell
motsi.py Diff [-h] -m MTI [MTI ...] [-t THRESHOLD] [-o OUTPUT] [--fillna FILL_NA] [--rep_weight REP_WEIGHT]
                     [--input_value INPUT_VALUE]
```

required arguments:

-   `-m` : List of input .mti files. Multiple files represent multiple replications.

options:

-   `-t`. Stable Differential Gene STFC thresholds (Default: 0.5)
-   `-o`: Output dir. (Default: DifferentialResult)
-   `--fillna` : Fill NA value with 0. (Default: True)
-   `--rep_weight`: The weight for replications when calculating STD (Default: 0.5)
-   `--input_value`: You can choose between inputting RawValues and Log2FoldChange. {raw,LFC}  (Default: raw)

## Output

After running `Diff`, you will obtain the following files:

#### **all_{target}.tsv**

A data table containing the computational results for all targets. The columns mean:

- base columns: Target information 
- value columns: The mean value between all replications.
- STS: The raw regulation score for each replication.
- UpPvalue: The p-value for considering a target as an **Up-regulated Target**
- DownPvalue: The p-value for considering a target as an **Down-regulated Target**
- *AllMark*: A score used for ranking all targets, including both Up-regulated and Down-regulated ones. It is used in the heatmap generated by this function.

#### up{target}.tsv / down{target}.tsv

The columns are identical to those in `all_{target}.tsv`. They respectively contain the filtered Up-regulated and Down-regulated targets.

#### All/Up/Down{target}_Heatmap.pdf 

Heatmaps used to visualize the three aforementioned data tables. The values displayed represent the average of all replicate input values. These averages are transformed by calculating `Log10(value + 1)`, followed by Z-score normalization for each target (i.e., each row).

